### ISSUE TITLE
Merge `docs` and `source` repositories

### ISSUE THREAD

**Issue by daquinteroflex** [flexcompute/tidy3d#1231]
We'd like to merge the documentation with the source code to have a more maintainable documentation environment and configuration. However, we don't want to give `rtd` organization access. One way to solve this is to implement `rtd` web hooks as described [on these instructions](https://docs.readthedocs.io/en/stable/guides/setup/git-repo-manual.html) The strategy to perform the actual repository merge could follow something like [this blog post](https://gfscott.com/blog/merge-git-repos-and-keep-commit-history/)

This issue is to track progress integrating the docs and source code, and whatever arises from doing that.

**Comment by daquinteroflex** [comment]
A few things to note, each of those pull requests contains the non-conflicting repository status of the docs and source code, **in my own personal forks**. The idea is that by changing the remote now of the docs repo (which again the remotes are my own personal forks), it should be possible to rebase all the git history of the docs into the main repository. 

It is important to mention we should be able to support progressive integration from the docs to the source until whenever we want, as long as all the work is contained within the `docs/` directory.

Whenever we want to integrate with the `develop` branch, then we just should need to merge https://github.com/flexcompute/tidy3d/pull/1234

**Comment by tylerflex** [comment]
>  Are we happy to restructure the docs/ folder so that it only contains documentation inside there?

Yea I think that should be fine.

>  Are we happy with this merge approach in order to maintain git history?
>  Are we happy to stick to the source code tidy3d license?

Maybe @momchil-flex has some thoughts on license and git history.

**Comment by momchil-flex** [comment]
> > Are we happy with this merge approach in order to maintain git history?
> > Are we happy to stick to the source code tidy3d license?
> 
> Maybe @momchil-flex has some thoughts on license and git history.

License sounds fine. Re the merge approach, is that explained in detail somewhere? It is not clear to me from just the messages here.

**Comment by daquinteroflex** [comment]
Sounds good! So the merge approach is in development still. The idea
currently is to move all example notebooks to a separate repo, and merge
the docs/source code to improve shared packaging and development. This
should be nearly tiny impact on the current web examples deployment
approach, and should streamline the documentation builds.

**Comment by momchil-flex** [comment]
The part I'm still not sure about is that you seem to be saying this can preserve the docs history? Would that mean all of the docs commits applied on top of the current tidy3d frontend? But somehow with notebooks removed?

**Comment by daquinteroflex** [comment]
There’s a few ways to do this. Check out this pull request
https://github.com/flexcompute/tidy3d/pull/1234
. We can first merge them in with all the history and then remove them when
we add the examples repo submodule. We can also just start from a clean
slate if we don’t want to preserve history.

We can move things around to convert it just in a notebook repo as well and
then plug that into the submodule later. Or just make a new notebook repo
from the state it’s in now. There’s a few options on what we could do.

Also open to ideas on how to preserve things best. What do you prefer?

**Comment by daquinteroflex** [comment]
Worth adding we could rebase the docs history into the frontend if that’s
how we wanted to track the history if we were to go with that option

**Comment by momchil-flex** [comment]
I'd say probably no need to preserve the docs history in the frontend. We have it in the docs repo if ever needed.

**Comment by daquinteroflex** [comment]
So I've been thinking how to implement this at the moment. Before, when merging the whole history, it kind of created two remote links between the two repositories, which meant that they could be both updated progressively until the day we decided to modify the docs one into just notebooks. In the sense that all commits could be propagated along to the new combined repository from both `tidy3d` and `tidy3d-docs` repositories.

I'm still investigating now how to implement this without necessarily keeping development history, but letting it progressively be integrate-able. Just because it might still take a few more days to sort out the rest of the documentation porting and writing the docs for the rest of the team on how to set up.

Another approach is that we maintain the history in order to maintain the progressive integration features, but we remove the notebooks on the new combined repo, and then we just modify the docs to notebook repo. The benefit of this is that the docs files that have nothing to do with the notebooks retain their history, and the docs repo maintains all its notebook history too. The real motivation though is the progressive integration aspect of this though.

I'm thinking of how to sort this and will update this message whenever I've found a potential solution for progressive integration without commit history mapping.
